src/GasEstimator.ts

```diff
@@ -0,0 +1,262 @@
+import {
+    constants,
+    ContractInteractor,
+    DeployTransactionRequest,
+    ERC20Token,
+    EstimateGasParams,
+    estimateMaxPossibleRelayCallWithLinearFit,
+    RelayTransactionRequest
+} from '@rsksmart/rif-relay-common';
+import {
+    DeployRequest,
+    DeployRequestStruct,
+    ForwardRequest,
+    RelayRequest
+} from '@rsksmart/rif-relay-contracts';
+import BigNumber from 'bignumber.js';
+import { toWei } from 'web3-utils';
+
+const SUBSIDY = BigNumber(12000);
+
+/**
+ * Estimates the max possible gas consumed by relaying a transaction using either a linearFit/standard estimation
+ * @param contractInteractor object containing the contractInteractor
```

Comment:
Generally speaking, the code should be understandable without comments. If the code does need commenting, however, they should be descriptive. Otherwise they have little value and in fact just spam the code, making it less readable. So if I see a comment with a param description, I expect to see at least a brief definition of the parameter.
`contractInteractor object containing the contractInteractor` does not tell me anything new that I did not know from looking at the function signature.

Reply:
@franciscotobar I think this is still valid

Reply:
Im not sure what approach should I use, I added an extra description but for example, BigNumber and BN, are external libraries that did not required an explanation about what are those, since are only used inside the functions. I was following the same assumption with the contractInteractor.